### PR TITLE
Enhance Notification : short/long notification type

### LINKED ISSUE #58: Customizable (or one-liner) notifications
**Is your feature request related to a problem? Please describe.**
Just a nitpick, but the notifications for events such as "deployment created" are pretty large in size.  

**Describe the solution you'd like**
You could probably easily fit all the info on one (Kind/Name/Namespace/Message/Cluster) into a single line, e.g., "${cluster}/${namespace} - ${kind} ${name}: ${message}".  If there are additional fields you could probably let the user customize this in the ConfigMap.

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by aananthraj**
##### ISSUE TYPE

 - Feature Pull Request

##### SUMMARY

This commit, (updated)
- enables sending short/long event notifications to channels
- adds an optional filed "notiftype" under channel config to change between short (default) and long notif types
- adds Message() method to event object for creating brief messages (to use across handlers)
- BugFix: enable/disable recommendations through the config file

Fixes #58 

##### Preview
preview

### CODE REVIEW

**Comment by PrasadG193** [comment]
Hi @aananthraj,
Thanks for the PR. This looks awesome! Just a minor comment on the message format, can we also highlight `cluster name` in the notification?

File: pkg/events/events.go [hunk 1]
```diff
@@ -183,3 +183,45 @@ func New(object interface{}, eventType config.EventType, kind string) Event {
...
+// included as a part of event package to use across handlers.
+func (event *Event) Message() (msg string) {
+	message := ""
+	if len(event.Messages) > 0 {
+		for _, m := range event.Messages {
+			message = message + m
+		}
+	}
+	switch event.Type {
+	case "create", "delete", "update":
```
[the remark this answers is not in the document]
Comment (aananthraj) [reply pkg/events/events.go]: resolved it

File: pkg/notify/mattermost.go [hunk 2]
```diff
@@ -44,77 +45,93 @@ func NewMattermost(c *config.Config) (Notifier, error) {
...
-		{
-			Title: "Name",
-			Value: event.Name,
-			Short: true,
-		},
-	}
+	var fields []*model.SlackAttachmentField
+
+	switch m.NotifType {
+	case "brief":
```
[the remark this answers is not in the document]
Comment (aananthraj) [reply pkg/notify/mattermost.go]: resolved it

**Comment by aananthraj** [comment]
any more suggestions..?

**Comment by PrasadG193** [comment]
@aananthraj, we need to add configuration related documentation in config files - `helm/botkube/values.yaml`, `deploy-all-in-one*.yaml` and `./config.yaml`. Otherwise, there is no way for the user to know about this feature.
Also, I think instead of `brief`, we should set `notiftype: long/short`. We should set `notiftype: short` as a default value across all the config files, and if the user wants, they can switch to `notiftype: long`.

Let me know what you think.

**Comment by aananthraj** [comment]
> Otherwise, there is no way for the user to know about this feature.

Yes @PrasadG193 , was thinking about updating docs as well. Will do it.

> we should set notiftype: long/short. We should set notiftype: short as a default value across all the config files,

Sounds good, will make the required changes and update here.

**Comment by PrasadG193** [comment]
@aananthraj, Are we not adding Recommendations in the notifications when configured with `short` type? We should add it. Otherwise, it will defeat the purpose of having filters

**Comment by aananthraj** [comment]
@PrasadG193, doing so there would be no significance between both short/long notif types. Long type will not be used at all (as we have all infos in a short way) which questions the existance of Long NotifType. Is this ok..?

**Comment by PrasadG193** [comment]
@aananthraj with `short` notification type, we do optimize a lot of space by combining name, kind, namespace and clustername into a single line. If anyone still wants to optimize more, they can always set `recommendations: false` in the config file.

**Comment by aananthraj** [comment]
@PrasadG193 requested changes were implemented. I guess it's ready now. Let's get it merged.

**Comment by PrasadG193** [comment]
Great work, @aananthraj !! :tada:

**Comment by aananthraj** [comment]
Thanks @PrasadG193 for the extended support 👍

**Pull request merged**